src/AI/CoreLogic.pm

```diff
@@ -1904,51 +1896,55 @@ sub processAutoBuy {
 			return unless (timeOut($args->{'recv_buyList_time'}, $timeout{ai_buyAuto_wait_before_buy}{timeout}));
 		}
 		
-		my @buyList;
-		
-		my $item = $storeList->getByName( $config{"buyAuto_".$args->{lastIndex}} );
-		
-		if (defined $item) {
-			$args->{'nameID'} = $item->{nameID};
-		}
+		return if !$args->{lastIndex};		
 		
-		if (!exists $args->{'nameID'}) {
-			$args->{index_failed}{$args->{lastIndex}} = 1;
-			error "buyAuto index ".$args->{lastIndex}." (".$config{"buyAuto_".$args->{lastIndex}}.") failed, item doesn't exist in npc sell list.\n", "npc";
-			
-		} else {
-			my $maxbuy = ($config{"buyAuto_".$args->{lastIndex}."_price"}) ? int($char->{zeny}/$config{"buyAuto_$args->{index}"."_price"}) : 30000; # we assume we can buy 30000, when price of the item is set to 0 or undef
-			my $needbuy = $config{"buyAuto_".$args->{lastIndex}."_maxAmount"};
+		my @buyList;
+		my $item;
+		my $zenyleft = $char->{zeny};
+
+		for(my $i = 0; exists $config{"buyAuto_$i"}; $i++){
+			next if (($config{"buyAuto_".$args->{lastIndex}."_npc"} ne $config{"buyAuto_".$i."_npc"})
+				  || (($i != $args->{lastIndex}) && !checkItemBuyNeed($i,$zenyleft)));
+			$item = $storeList->getByName( $config{"buyAuto_".$i} );
+			if (defined $item){
+				$args->{'nameID'} = $item->{nameID};
+			}
+			if (!exists $args->{'nameID'}){
+				$args->{index_failed}{$i} = 1;
+				error "buyAuto index ".$i." (".$config{"buyAuto_".$i}.") failed, item doesn't exist in npc sell list.\n", "npc";
+			}else{
+				my $maxbuy = ($config{"buyAuto_".$i."_price"}) ? int($zenyleft/$config{"buyAuto_".$i."_price"}) : 30000; # we assume we can buy 30000, when price of the item is set to 0 or undef
```

Comment:
i think it is a little danger to set the amout to be the max that money can buy
so if a new user forgot do put maxAmaount, and it is a char with a LOT of zeny
openkore will try to buy 30000 of the item? very dangerous